vue_dynamic-list-of-posts/src/components/PostPreview.vue

```diff
@@ -0,0 +1,119 @@
+<script>
+import CommentForm from "./CommentForm.vue";
+import Comment from "./Comment.vue";
+import Loader from "./Loader.vue";
+import { getPostComments, addCommentPost, deleteComment } from '../api/comments';
+
+export default {
+  name: 'PostPreview',
+  components: {
+    CommentForm,
+    Comment,
+    Loader,
+  },
+  props: {
+    post: Object,
+  },
+  emits: ['deletePost', 'editPost'],
+  data() {
+    return {
+      showCommentForm: false,
+      comments: [],
+      loading: false,
+    }
+  },
+  mounted() {
+    this.loadComments();
+  },
+  watch: {
+    post: {
+      handler(newValue, oldValue) {
+        if (newValue.id !== oldValue.id) {
+          this.loadComments();
+        }
+      },
+      deep: true,
+    }
+  },
+  methods: {
+    toggleCommentForm() {
+      this.showCommentForm = !this.showCommentForm;
+    },
+    loadComments() {
+      this.loading = true;
+      getPostComments(this.post.id)
+      .then((res) => {
+        this.comments = res.data;
+      })
+      .finally(() => {
+        this.loading = false;
+      })
+    },
+    createComment(data) {
+      addCommentPost({...data, postId: this.post.id})
+        .then((res) => {
+          this.comments.push(res.data);
+        });
+    },
+    removeComment(commentId) {
+      const indexToRemove = this.comments.findIndex(comment => comment.id === commentId);
+
+      if (indexToRemove === -1) {
+        return;
+      }
+
+      const commentToRemove = this.comments[indexToRemove];
+      this.comments.splice(indexToRemove, 1);
+
+      deleteComment(commentId)
+        .then((res) => {
+          console.log('success', res.data);
+        })
+        .catch((error) => {
+          this.comments.splice(indexToRemove, 0, commentToRemove);
+          console.error('Failed to delete comment:', error);
+        });
```

Comment:
```suggestion
     this.comments = this.comments.filter(comment => comment.id !== commentId);
     
      deleteComment(commentId)
        .then((res) => {
          console.log('success', res.data);
        })
        .catch((error) => {
          this.comments.splice(indexToRemove, 0, commentToRemove);
          console.error('Failed to delete comment:', error);
        });
```

Maybe I missed something, but as I see we can simplify